src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticateResult.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// Contains the result of an Authenticate call
+    /// </summary>
+    public class AuthenticateResult
+    {
+        private AuthenticateResult() { }
+
+        /// <summary>
+        /// If a ticket was produced, authenticate was successful.
+        /// </summary>
+        public bool Succeeded
+        {
+            get
+            {
+                return Ticket != null;
+            }
+        }
+
+        /// <summary>
+        /// The authentication ticket.
+        /// </summary>
+        public AuthenticationTicket Ticket { get; private set; }
+
+        // REVIEW: should we also remember AuthenticationScheme?
+
+        /// <summary>
+        /// Gets the claims-principal with authenticated user identities.
+        /// </summary>
+        public ClaimsPrincipal Principal => Ticket?.Principal;
+
+        /// <summary>
+        /// Additional state values for the authentication session.
+        /// </summary>
+        public AuthenticationProperties Properties => Ticket?.Properties;
+
+        /// <summary>
+        /// Holds failure information from the authentication.
+        /// </summary>
+        public Exception Failure { get; private set; }
+
+        /// <summary>
+        /// Indicates that stage of authentication was directly handled by user intervention and no
+        /// further processing should be attempted.
+        /// </summary>
+        public bool Handled { get; private set; }
+
+        /// <summary>
+        /// Indicates that there was no information returned for this authentication scheme.
+        /// </summary>
+        public bool Nothing { get; private set; }
+
+        public static AuthenticateResult Success(AuthenticationTicket ticket)
+        {
+            if (ticket == null)
+            {
+                throw new ArgumentNullException(nameof(ticket));
+            }
+            return new AuthenticateResult() { Ticket = ticket };
+        }
+
+        public static AuthenticateResult Handle()
+        {
+            return new AuthenticateResult() { Handled = true };
+        }
+
+        public static AuthenticateResult None()
```

Comment:
Skip makes sense when its part of the pipeline, like for handle request in the remote case.  It doesn't really make sense when you ask for an identity to get skip...  Empty or Nothing is really what we want here, basically its when there's No cookie, or no token...  NotFound, NoPrincipal also work